exercises/transpose/src/example/java/Transpose.java

```diff
@@ -0,0 +1,41 @@
+public class Transpose {
+    public String transpose(String toTranspose) {
+
+        StringBuilder transposedForm = new StringBuilder();
+        String[] rows = toTranspose.split("\n");
+
+        int maxRowSize = 0;
+
+        for (int i = 0; i < rows.length; i++) {
+            if (rows[i].length() > maxRowSize) {
+                maxRowSize = rows[i].length();
+            }
+        }
+
+        padRows(rows);
+
+        for (int newRowOldCol = 0; newRowOldCol < maxRowSize; newRowOldCol++) {
+            for (int newColOldRow = 0; newColOldRow < rows.length; newColOldRow++) {
+                if (newRowOldCol < rows[newColOldRow].length()) {
+                    transposedForm.append(rows[newColOldRow].charAt(newRowOldCol));
+                }
+            }
+
+            if (newRowOldCol != maxRowSize - 1) {
```

Comment:
From what I can see then `maxRowSize` is only being used to check whether you've reached the last new row? To simplify that you could store each new row in an array and then at the end do `String.join("\n". newRows)`. You definitely don't have to do that though, it's just a suggestion :)